### DISCUSSION TITLE
fetch storage from multiple folders in a bucket

### DISCUSSION THREAD

**Discussion by danbeck79** [supabase/supabase#17762] [Questions] [answered]
The scenario: 

I have 3 folders inside a bucket: `images` `videos` `files`.

On the FE i want to show what assets are in the bucket (12 at a time, with a load more). the reason for the folders is that sometimes i only want to display 12 images, videos or files at a time (felt like the simplest method). but there will be a time where i want to mix all three in 12 at a time basis. What would be the recommended approach here?

**Comment by GaryAustin1** [comment]
Are you wanting the actual images/files or just their names?  
If you are just wanting to list the names I would just go right to the storage.objects table with a database call with .schema('storage') set, or use an rpc.  The storage .list files can only do one folder at a time.

**Reply by danbeck79** [reply]
I want the files too as I want to display thumbnails where possible. Would you still recommend via rpc

**Reply by GaryAustin1** [reply, marked as answer]
You have to download files one at a time from the Storage APi with either the publicURL or signedURL, or download().
So you have to get the list of files you want thru whatever method and then download them in a loop or similar
.

**Reply by danbeck79** [reply]
Brilliant, thanks for your help
